From f864c65a469fc4f53337ba09157693bf90eb9c66 Mon Sep 17 00:00:00 2001
From: Sergio Pena <sergio.pena@cloudera.com>
Date: Thu, 7 Jul 2016 16:30:20 -0500
Subject: [PATCH 0663/1164] CLOUDERA-BUILD: CDH-42090:
 getNumPartitionsByFilter fail on direct-sql due
 to datanucleus old version

This fixes an issue related with the backport of HIVE-13884 (CDH-41944). The next Hive 2.x should have
this fixed, so it is not necessary to rebase it.

Change-Id: Iba8a9fb8e1c0774ef69e653059afeadd92169bb1
---
 .../hadoop/hive/metastore/MetaStoreDirectSql.java  |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java b/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java
index fb4ee61..88fbb7d 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java
@@ -838,11 +838,15 @@ public int getNumPartitionsViaSqlFilter(SqlFilterForPushdown filter) throws Meta
 
     long start = doTrace ? System.nanoTime() : 0;
     Query query = pm.newQuery("javax.jdo.query.SQL", queryText);
-    @SuppressWarnings("unchecked")
-    int sqlResult = extractSqlInt(query.executeWithArray(params));
+
+    List<Integer> result = executeWithArray(query, params, queryText);
+    if (result == null || result.size() == 0) {
+      throw new MetaException("Could not get number of partitions from Direct SQL.");
+    }
+
     long queryTime = doTrace ? System.nanoTime() : 0;
     timingTrace(doTrace, queryText, start, queryTime);
-    return sqlResult;
+    return extractSqlInt(result.get(0));
   }
 
 
-- 
1.7.9.5

